File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer, tmp;
    for(int i=0;i<score.size();i++){

        tmp.push_back( score[i] );
        sort(tmp.begin(), tmp.end(), greater<>());

        if(tmp.size() <= k) answer.push_back(tmp.back());
        else answer.push_back(tmp[k-1]);
    }

    return answer;
}

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        int pos = 0;
        Vector<Integer> v = new Vector<>();

        for(int s : score){
            v.add(s);
            Collections.sort(v, (Integer a, Integer b)-> b - a);

            if(v.size() <= k ) answer[pos++] = v.lastElement();
            else answer[pos++] = v.get(k-1);

        }
        return answer;
    }
}